### ISSUE TITLE
UP007 should leave type aliases alone

### ISSUE THREAD

**Issue by aberres** [charliermarsh/ruff#2981]
Given the following snippet, `ruff` should not throw an error. At least this is the behavior of `pyupgrade`.

There are cases where one still needs an explicit `Optional`. See https://github.com/tiangolo/typer/issues/348 / https://github.com/tiangolo/typer/issues/348#issuecomment-1337460105 for an example.

### Code

```
from __future__ import annotations

from typing import Optional

StrOrNone = Optional[str]
```

### Output

```
up007.py:6:13: UP007 [*] Use `X | Y` for type annotations
Found 1 error.
```

**Comment by charliermarsh** [comment]
Ah that's interesting. I think we used to _not_ change this (following pyupgrade), but I didn't see a good reason not to, and changed it. We can revert.

Note that pyupgrade _will_ fix these cases:

```py
from __future__ import annotations

from typing import List

StrOrNone = List[str]
```

**Comment by twoertwein** [comment]
might be wrong: I think this was just a limitation of mypy<1.0

**Comment by aberres** [comment]
To have some more context, the use case for me is like this:

![image](https://user-images.githubusercontent.com/20811121/219692758-215a1a37-a8f4-4ee2-87d8-3512bf9767f6.png)

Without the indirection `pyupgrade` modernizes the code as expected, and `typer` fails.

**Comment by charliermarsh** [comment]
Yeah that makes sense.

Is there any sense for when https://github.com/tiangolo/typer/pull/522 would be merged?
